apps/prairielearn/src/lib/editors.ts

```diff
@@ -502,112 +503,115 @@ export abstract class Editor {
       }
     }
   }
+}
 
-  /**
-   * Get all existing shortnames, recursing on nonempty directories that do not contain
-   * an ".info" file.
-   * @param rootDirectory Directory to start searching from.
-   * @param infoFile Name of the info file, will stop recursing once a directory contains this.
-   */
-  async getExistingShortNames(rootDirectory: string, infoFile: string) {
-    const files: string[] = [];
-    const walk = async (relativeDir) => {
-      const directories = await fs.readdir(path.join(rootDirectory, relativeDir)).catch((err) => {
-        // If the directory doesn't exist, then we have nothing to load
-        if (err.code === 'ENOENT' || err.code === 'ENOTDIR') {
-          return [] as string[];
-        }
-        throw err;
-      });
-
-      // For each subdirectory, try to find an Info file
-      await async.each(directories, async (dir) => {
-        // Relative path to the current folder
-        const subdirPath = path.join(relativeDir, dir);
-        // Absolute path to the info file
-        const infoPath = path.join(rootDirectory, subdirPath, infoFile);
-        const hasInfoFile = await fs.pathExists(infoPath);
-        if (hasInfoFile) {
-          // Info file exists, we can use this directory
-          files.push(subdirPath);
-        } else {
-          // No info file, let's try recursing
-          await walk(subdirPath);
-        }
-      });
-    };
-
-    await walk('');
-    debug('getExistingShortNames() returning', files);
-    return files;
-  }
-
-  getNamesForCopy(
-    oldShortName: string,
-    shortNames: string[],
-    oldLongName: string | null,
-    longNames: string[],
-  ): { shortName: string; longName: string } {
-    function getBaseShortName(oldname: string): string {
-      const found = oldname.match(new RegExp('^(.*)_copy[0-9]+$'));
-      if (found) {
-        return found[1];
-      } else {
-        return oldname;
+/**
+ * Get all existing shortnames, recursing on nonempty directories that do not contain
+ * an "info" file.
+ *
+ * TODO: does this need to use the disk? Why can't we query the database instead?
```

Comment:
We don't have to do anything in this PR, but I can't figure out why we're using disk for this stuff. Is there a reason we can't query the database to get this information? If we can figure out a reason, I'd love to document it for our future selves.